SPIFlashLogger.class.nut

```diff
@@ -485,29 +712,38 @@ class SPIFlashLogger {
         }
         _disable();
 
-        _at_pos = 0;
-        _at_sec = best_sec;
-        _next_sec_id = best_id+1;
+        // handle ID overflow use-case
+        if (lastSector.id - firstSector.id >= (0xFFFF - _sectors))
```

Comment:
not sure I understand this statement, especially the following assignment. can you please elaborate?

Reply:
I'm afraid that I do not remember the logic of this statements too. (On the latest codebase 0xFFFF replaced with 0x7FFFf...)
We need to catch an overflow use-case here
So, if last.id == 0x7FFF.. and first.id == 2  then last-first >= 0x7fff... -sectors(for example we have  15-17 sectors).
So, everything looks logically. If we have a correct overflow then we take the firstSector as start position, otherwise we take the sector with the highest ID